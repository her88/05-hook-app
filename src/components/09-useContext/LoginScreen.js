import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const LoginScreen = () => {

    const {setUser} = useContext(UserContext);

    const userSession = {
        id: 123,
        name: 'Hernan',
    }

    return (
        <div>
            <h1>LoginScreen</h1>
            <hr />

            <button className="btn btn-primary" onClick={ () => setUser(userSession)}>
                Login
            </button>
        </div>
    )
}
