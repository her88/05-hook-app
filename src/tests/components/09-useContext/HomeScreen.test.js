import { mount } from "enzyme";
import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from "../../../components/09-useContext/UserContext";


describe('Pruebas en <HomeScreen />', () => {

    const user = {
        name: 'Hernan',
        email: 'her@gmail.com'
    }

    const wrapper = mount(
        <UserContext.Provider value={{user}}>
            <HomeScreen />
        </UserContext.Provider>
    );

    test('se debe mostrar correctamente el componente ', () => {
        
        expect(wrapper).toMatchSnapshot();
    });
    
    
});
