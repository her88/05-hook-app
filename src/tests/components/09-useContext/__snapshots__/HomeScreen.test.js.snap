// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomeScreen /> se debe mostrar correctamente el componente  1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "Hernan",
   "email": "her@gmail.com"
}
  </pre>
</div>
`;
